fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,125 @@
+    std::cout << "new SS file name: " << fileName << std::endl;
+    deleteFile(fileName);
+    state IKeyValueStore* data = openKVStore(storeType, fileName, recruited.id(), 0);
+
+    state Tag tag = Tag(1, 1);
+
+    state ReplyPromise<InitializeStorageReply> storageReady;
+
+    // Initialize dbInfo with mocked cursor and mocked loy system.
+    Arena arena;
+
+    MutationRef mutation(arena,
+                         MutationRef::SetValue,
+                         StringRef(std::string("Hello")),
+                         StringRef(std::string("World")));
+
+    VectorRef<Tag> tags;
+	tags.emplace_back(arena, tag);
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    MockPeekCursor::MessageAndTags messageAndTags;
+	messageAndTags.message = BinaryWriter::toValue(mutation, IncludeVersion());
+	messageAndTags.tags = tags;
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    VectorRef<MockPeekCursor::MessageAndTags> messagesInOnVersion;
+    messagesInOnVersion.emplace_back(arena, messageAndTags);
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    VectorRef<MockPeekCursor::OneVersionMessages> allVersionMessages;
+    allVersionMessages.emplace_back(arena,
+	                                8, // version
+                                    messagesInOnVersion);
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    Reference<MockPeekCursor> mockPeekCursor = makeReference<MockPeekCursor>(allVersionMessages, arena);
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+	// TODO: Use Reference API?
+    state MockLogSystem *mockLogSystem = new MockLogSystem();
+    mockLogSystem->cursor = mockPeekCursor.castTo<ILogSystem::IPeekCursor>();
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    LogSystemConfig logSystemConfig;
+    logSystemConfig.logSystemType = LogSystemType::mock;
+    logSystemConfig.mockLogSystem = mockLogSystem;
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+    ServerDBInfo dbInfoBuilder;
+    dbInfoBuilder.recoveryState = RecoveryState::ACCEPTING_COMMITS;
+    dbInfoBuilder.logSystemConfig = logSystemConfig;
+    state Reference<AsyncVar<ServerDBInfo>> dbInfo = makeReference<AsyncVar<ServerDBInfo>>(dbInfoBuilder);
+    // Initialize dbInfo done.
+    std::cout << "wawawa storageserver test " << __LINE__ << std::endl;
+
+
+    std::cout << "before storageServer" << std::endl;
+    state Future<Void> s = storageServer(data, recruited, tag, storageReady, dbInfo, folder);
+    std::cout << "wait storageServer" << std::endl;
+
+    std::vector<Future<Void>> actors;
+	actors.emplace_back(s);
+    // Timeout after 1 second.
+    std::cout << "before delay" << std::endl;
+    wait(delay(0.1));
+    std::cout << "after delay" << std::endl;
+
+	//TODO: why this not work?
```

Comment:
This should work. What problem do you see?

Reply:
I think I still have the problem but unrelated to this code now.